### PR TITLE
Implement byte mask load

### PR DESCRIPTION
**Pull request by gita-omr**
- when a byte mask is loaded from a byte array the true value is represented as 1.
- expand the least significant bit so that instructions like xxsel work correctly.

### CODE REVIEW

**Comment by gita-omr** [comment]
@zl-wang please kindly review

File: compiler/p/codegen/OMRTreeEvaluator.cpp [hunk 1]
```diff
@@ -3098,6 +3104,17 @@ TR::Register *OMR::Power::TreeEvaluator::vloadEvaluator(TR::Node *node, TR::Code
 
    TR::LoadStoreHandler::generateLoadNodeSequence(cg, dstReg, node, opcode, 16, true);
```
Comment (zl-wang) [inline compiler/p/codegen/OMRTreeEvaluator.cpp:3106]: for this functionality, i don't think you need `vexpandbm`.  why not use `vsrab` instead, in place of `vexpandbm`?  `vsrab` is available at least from POWER7.

File: compiler/p/codegen/OMRTreeEvaluator.cpp [hunk 2]
```diff
@@ -3098,6 +3104,17 @@ TR::Register *OMR::Power::TreeEvaluator::vloadEvaluator(TR::Node *node, TR::Code
...
 
+   if (expandOpcode != TR::InstOpCode::bad &&
+       node->getDataType().getVectorElementType() == TR::Int8)
```
Comment (zl-wang) [inline compiler/p/codegen/OMRTreeEvaluator.cpp:3108]: such that, i have a further question: what happened running the same IL tree on pre-POWER10, i.e. the same `vload` node?

**Review by zl-wang — CHANGES_REQUESTED** [review]

File: compiler/p/codegen/OMRTreeEvaluator.cpp (hunk 2 above)
Comment (gita-omr) [reply compiler/p/codegen/OMRTreeEvaluator.cpp:3105]: Whoever generates mask load should've been doing the checking. But not problem now with vsrab.

Comment (zl-wang) [reply compiler/p/codegen/OMRTreeEvaluator.cpp:3105]: somewhere in the common code, do you mean that now you can generate this mask load on more platforms (previously for P10 only)?  do I expect to see this checking modification in this PR as well?

Comment (gita-omr) [reply compiler/p/codegen/OMRTreeEvaluator.cpp:3105]: The P10 check was missing but not needed now.

Comment (zl-wang) [reply compiler/p/codegen/OMRTreeEvaluator.cpp:3105]: great.

**Review by zl-wang — APPROVED** [review]
LGTM now.

**Comment by zl-wang** [comment]
@dsouzai please review/approve/merge.

**Comment by gita-omr** [comment]
Squashed commits.

**Comment by dsouzai** [comment]
jenkins build plinux,aix

**Pull request merged**